static/js/components/AssignmentForm.jsx

```diff
@@ -0,0 +1,155 @@
+}
+
+function makeMenuItem(observingRun, instrumentList, telescopeList, groups) {
+  const render_string = observingRunTitle(observingRun, instrumentList, telescopeList, groups);
+  return (
+    <MenuItem value={observingRun.id} key={observingRun.id.toString()}>
+      {render_string}
+    </MenuItem>
+  );
+}
+
+const AssignmentForm = ({ obj_id, observingRunList }) => {
+  const dispatch = useDispatch();
+  const { instrumentList } = useSelector((state) => state.instruments);
+  const { telescopeList } = useSelector((state) => state.telescopes);
+  const groups = useSelector((state) => state.groups.all);
+
+  const upcomingRuns = observingRunList.filter((observingrun) => (
+    Date.parse(observingrun.ephemeris.sunrise_utc) >= Date.now()
+  ));
+
+  const { handleSubmit, getValues, reset, register, control } = useForm();
+
+  const useStyles = makeStyles((theme) => ({
+    formControl: {
+      margin: theme.spacing(1),
+      minWidth: 120
+    }
+  }));
+  const classes = useStyles();
+
+  if (upcomingRuns.length === 0) {
+    return (
+      <b>
+        No upcoming observing runs to assign target to...
+      </b>
+    );
+  }
+
+  const initialFormState = {
+    comment: "",
+    run_id: upcomingRuns.length > 0 ? upcomingRuns[0].id : null,
+    priority: "1",
+    obj_id
+  };
+
+  const onSubmit = () => {
+    const formData = {
+      // Need to add obj_id, etc to form data for request
+      ...initialFormState,
+      ...getValues({ nest: true })
+    };
+    dispatch(Actions.submitAssignment(formData));
+    reset(initialFormState);
+  };
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <h3>
+          Assign Target to Observing Run
+        </h3>
+        <div>
+          <FormControl className={classes.formControl}>
+            <InputLabel id="assignmentSelectLabel">
+              Choose Run
+            </InputLabel>
+            <Controller
+              as={Select}
+              labelId="assignmentSelectLabel"
+              name="run_id"
+              control={control}
+              rules={{ required: true }}
+              defaultValue={upcomingRuns.length > 0 ? upcomingRuns[0].id : null}
+            >
+              {upcomingRuns.map((observingRun) => makeMenuItem(
+                observingRun, instrumentList, telescopeList, groups
+              ))}
+            </Controller>
+          </FormControl>
+          <FormControl className={classes.formControl}>
+            <InputLabel id="prioritySelectLabel">
+              Priority
+            </InputLabel>
+            <Controller
+              as={Select}
+              labelId="prioritySelectLabel"
+              defaultValue="1"
+              name="priority"
+              control={control}
+              rules={{ required: true }}
+            >
+              {
+                ["1", "2", "3", "4", "5"].map((prio) => (
```

Comment:
Does this need to be hardcoded? Or should the set of possibilities be grabbed from the state? I can imagine a different SP instance using different priority nomenclature.

Reply:
Deferred to #603